beacon.go

```diff
@@ -17,110 +17,134 @@
+		return false, fmt.Errorf("creating bigquery client: %v", err)
 	}
-	if allele == "" {
-		return nil, "", errors.New("missing allele")
+	it, err := bqclient.Query(query).Read(ctx)
+	if err != nil {
+		return false, fmt.Errorf("querying database: %v", err)
 	}
 
-	return search, allele, nil
+	var result struct {
+		Count int
+	}
+	if err := it.Next(&result); err != nil {
+		return false, fmt.Errorf("reading query result: %v", err)
+	}
+	return result.Count > 0, nil
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	c := appengine.NewContext(r)
-	search, allele, err := requestToSearch(r)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+func validateServerConfig() error {
+	if config.projectID == "" {
+		return fmt.Errorf("%s must be specified", projectKey)
+	}
+	if config.tableId == "" {
+		return fmt.Errorf("%s must be specified", bqTableKey)
 	}
+	return nil
+}
 
-	client, err := google.DefaultClient(c, genomics.GenomicsReadonlyScope)
-	if err != nil {
-		http.Error(w, "Invalid server configuration", http.StatusInternalServerError)
+func parseInput(r *http.Request) (string, string, int64, error) {
+	refName := r.FormValue("chromosome")
+	if refName == "" {
+		return "", "", 0, errors.New("missing chromosome name")
 	}
-	genomicsService, err := genomics.New(client)
+	allele := r.FormValue("allele")
+	if refName == "" {
+		return "", "", 0, errors.New("missing allele")
+	}
+	coord, err := strconv.ParseInt(r.FormValue("coordinate"), 10, 64)
 	if err != nil {
-		http.Error(w, "Invalid server configuration", http.StatusInternalServerError)
+		return "", "", 0, fmt.Errorf("parsing coordinate: %v", err)
 	}
-	variantsService := genomics.NewVariantsService(genomicsService)
+	return refName, allele, coord, nil
+}
 
+func writeResponse(w http.ResponseWriter, exists bool) error {
 	type beaconResponse struct {
 		XMLName struct{} `xml:"BEACONResponse"`
 		Exists  bool     `xml:"exists"`
 	}
 	var resp beaconResponse
-
-	for {
-		searchResponse, err := variantsService.Search(search).Do()
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
-
-		for _, variant := range searchResponse.Variants {
-			if search.Start != variant.Start {
-				continue
-			}
-			if allele == variant.ReferenceBases {
-				resp.Exists = true
-			} else {
-				for _, base := range variant.AlternateBases {
-					if base == allele {
-						resp.Exists = true
-						break
-					}
-				}
-			}
-		}
-
-		if resp.Exists || searchResponse.NextPageToken == "" {
-			break
-		}
-		search.PageToken = searchResponse.NextPageToken
-	}
+	resp.Exists = exists
 
 	w.Header().Set("Content-Type", "application/xml")
 	enc := xml.NewEncoder(w)
 	enc.Indent("", "  ")
-	if err = enc.Encode(resp); err != nil {
-		http.Error(w, "Failed writing response", http.StatusInternalServerError)
+	if err := enc.Encode(resp); err != nil {
+		return fmt.Errorf("serializing response: %v", err)
 	}
-}
+	return nil
+}
```

Comment:
Did you run this through gofmt?  I thought it forced a newline at the end of the file.
[In either case, add a newline]

Reply:
forgot to. done